### DISCUSSION TITLE
Highlight lines of included file with docgen

### DISCUSSION THREAD

**Discussion by wiebekaai** [fuma-nama/fumadocs#474] [Ideas]
https://fumadocs.vercel.app/docs/headless/mdx/remark-docgen

Hey, any way I can highlight lines when including a file as code block? Or even better, only include certain lines?

**Comment by fuma-nama** [comment]
We use Shiki for code highlighting, you can use transformers inside your code: https://fumadocs.vercel.app/docs/ui/markdown#highlight-lines

**Reply by wiebekaai** [reply]
Thanks! I don't understand how to use it with docgen though. Here's what I tried:

```mdx
```json doc-gen:file
{
  "file": "../snippets/assets.liquid",
  "codeblock": {
    "lang": "liquid",
    "meta": "title=\"snippets/assets.liquid\" custom=\"{1}\" "
  }
}```
```

```mdx
```json doc-gen:file
{
  "file": "../snippets/assets.liquid",
  "codeblock": {
    "lang": "liquid",
    "meta": "title=\"snippets/assets.liquid\" {1}"
  }
}```
```

**Reply by fuma-nama** [reply]
I mean adding the comment like:
```
// [!code highlight]
```
To your code. This is called notation transformers in Shiki

**Reply by wiebekaai** [reply]
Ah, that makes sense. I tried using actual project code in the docs, but these comments break the code (.liquid files). Guess that's just not a good idea.

Thank you.

**Reply by fuma-nama** [reply]
I see, if your code isn't supported by notation transformers, you can use [meta highlight](https://shiki.matsu.io/packages/transformers#transformermetahighlight) instead. You can just pass it as meta string similar to your code above. 

This transformer isn't installed by default, so you might have to install the shiki/transformers package and add the transformer yourself.
Maybe I can open a PR to Shiki to fix this later
